encoding/kzg/verifier/v2/verifier.go

```diff
@@ -93,23 +95,32 @@ func (v *Verifier) newKzgVerifier(params encoding.EncodingParams) (*Parametrized
 }
 
 func (v *Verifier) VerifyBlobLength(commitments encoding.BlobCommitments) error {
-	return v.VerifyCommit(
+	return v.VerifyLengthProof(
 		(*bn254.G2Affine)(commitments.LengthCommitment),
 		(*bn254.G2Affine)(commitments.LengthProof),
 		uint64(commitments.Length),
 	)
 }
 
-// VerifyCommit verifies the low degree proof; since it doesn't depend on the encoding parameters
-// we leave it as a method of the KzgEncoderGroup
-func (v *Verifier) VerifyCommit(lengthCommit *bn254.G2Affine, lengthProof *bn254.G2Affine, length uint64) error {
+// VerifyLengthProof verifies the length proof (low degree proof).
+// See https://layr-labs.github.io/eigenda/protocol/architecture/encoding.html#validation-via-kzg
+// Since it doesn't depend on the encoding parameters, we leave it as a method of Verifier, not ParametrizedVerifier.
+func (v *Verifier) VerifyLengthProof(
+	lengthCommit *bn254.G2Affine, lengthProof *bn254.G2Affine, commitmentLength uint64,
+) error {
+	if !encoding.IsPowerOfTwo(commitmentLength) {
+		return fmt.Errorf("commitment length %d is not a power of 2", commitmentLength)
+	}
 
-	g1Challenge, err := kzg.ReadG1Point(v.kzgConfig.SRSOrder-length, v.kzgConfig.SRSOrder, v.kzgConfig.G1Path)
-	if err != nil {
-		return fmt.Errorf("read g1 point: %w", err)
+	commitmentLengthLog := bits.TrailingZeros64(commitmentLength)
+	if commitmentLengthLog > 28 {
+		return fmt.Errorf("commitment length %d is > max possible 2^28", commitmentLength)
 	}
+	// g1Challenge = [x^(SRSOrder - commitmentLength)]_1
+	// G1PowerOf2SRS contains the 28 hardcoded points that we need.
+	g1Challenge := srs.G1PowerOf2SRS[commitmentLengthLog]
```

Comment:
Where is this new logic coming from? Was it moved from somewhere else, or are you adding additional checks?

In particular, I'm not following the logic of the condition `bits.TrailingZeros64(commitmentLength) > 28`

```
	if !encoding.IsPowerOfTwo(commitmentLength) {
		return fmt.Errorf("commitment length %d is not a power of 2", commitmentLength)
	}

	commitmentLengthLog := bits.TrailingZeros64(commitmentLength)
	if commitmentLengthLog > 28 {
		return fmt.Errorf("commitment length %d is > max possible 2^28", commitmentLength)
	}
	// g1Challenge = [x^(SRSOrder - commitmentLength)]_1
	// G1PowerOf2SRS contains the 28 hardcoded points that we need.
	g1Challenge := srs.G1PowerOf2SRS[commitmentLengthLog]
```

Reply:
It's more clear now, but this seems like it might be more clever than it needs to be.

Wouldn't it be easier to just compare the commitmentLength to the maximum permitted value? Just a thought, feel free to ignore if there's a good reason to go with the clever solution